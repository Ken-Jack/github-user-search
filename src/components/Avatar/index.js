// @flow

import React from 'react';
import Image from 'components/Image';
import {connect} from 'react-fela';

type Props = {
  url: string,
  name: string | null,
  width: number,
  styles: Object,
};

function Avatar({url, name, width, styles}: Props) {
  const alt = name ? `${name}'s avatar` : 'User avatar';
  return (
    <Image
      className={styles.avatar}
      src={`${url}&s=${width * 2}`}
      alt={alt}
      width={width}
    />
  );
}

const styles = {
  avatar: () => ({
    borderRadius: '8px',
  }),
};

export default connect(styles)(Avatar);
