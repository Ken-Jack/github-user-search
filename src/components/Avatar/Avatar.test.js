import React from 'react';
import felaSnapshot from 'test-util/fela-snapshot';
import Avatar from './Avatar';

describe('Component: Avatar', () => {

  it('should render', () => {
    const component = felaSnapshot(
      <Avatar
        url="http://avatar.net"
        name="Simon"
        width={300}
      />
    );
    expect(component).toMatchSnapshot();
  });

});
