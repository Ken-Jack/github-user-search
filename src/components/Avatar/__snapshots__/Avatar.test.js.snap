// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Avatar should render 1`] = `
Object {
  "component": <div
    className="b"
    style={
      Object {
        "minHeight": 60,
      }
    }
  >
    <img
      alt="Simon's avatar"
      className="a g h i"
      onLoad={[Function]}
      src="http://avatar.net&s=600"
      theme={Object {}}
      width={300}
    />
    <svg />
  </div>,
  "styles": "
    .a{border-radius:8px}
    .b{position:relative}
    .c{left:50%}
    .d{top:50%}
    .e{position:absolute}
    .f{transform:translate(-50%, -50%)}
    .g{display:block}
    .h{min-width:100%}
    .i{height:auto}",
}
`;
